blocks/header/header.js

```diff
@@ -78,62 +100,138 @@ async function buildNav() {
   }
 }
 
+/**
+ * Sets up the autocomplete functionality for the search input.
+ * @returns {Promise<void>}
+ */
+async function setupAutocomplete() {
+  try {
+    const searchInput = await waitForElement('#navSearch');
+    const autocompleteList = await waitForElement('#autocomplete-list');
+
+    // Fetch data from various sheets
+    const communityResult = await getCommunitiesSheet();
+    const staffResult = await getStaffSheet();
+    const modelResult = await getModelsSheet();
+    const inventoryResult = await getInventorySheet();
+
+    // Format the data for autocomplete
+    const communityData = formatCommunities(communityResult.data);
+    const staffData = formatStaff(staffResult.data);
+    const modelData = formatModels(modelResult.data);
+    const inventoryData = formatInventory(inventoryResult.data);
+
+    // Combine all formatted data
+    const allSuggestions = [
+      ...communityData,
+      ...staffData,
+      ...modelData,
+      ...inventoryData,
+    ];
+
+    const handleInput = throttle(() => {
+      const value = searchInput.value.toLowerCase();
+      autocompleteList.innerHTML = '';
+
+      if (value.length < 2) return;
+
+      const filteredSuggestions = allSuggestions.filter((item) => item.display
+        .toLowerCase().includes(value));
+      filteredSuggestions.forEach((suggestion) => {
+        const item = document.createElement('div');
+        item.innerHTML = `<a href="${suggestion.path}" class="autocomplete-link">${suggestion.display}</a>`;
+        item.addEventListener('click', () => {
+          searchInput.value = suggestion.value;
+          autocompleteList.innerHTML = '';
+        });
+        autocompleteList.appendChild(item);
+      });
+    }, 300); // Adjust the throttle limit as needed
+
+    searchInput.addEventListener('input', handleInput);
+
+    document.addEventListener('click', (e) => {
+      if (e.target !== searchInput) {
+        autocompleteList.innerHTML = '';
+      }
+    });
+  } catch (error) {
+    console.error(error.message);
```

Comment:
console usage should cause failures in the build.

Reply:
removed